strawberry/alloptional.py

```diff
@@ -0,0 +1,23 @@
+from abc import ABCMeta
+from typing import Optional
+
+import strawberry
+
+
+class AllOptional(ABCMeta):
+    def __new__(cls, name: str, bases: tuple, namespaces: dict, **kwargs):
+        annotations = namespaces.get("__annotations__", {})
+        fields: list[str] = []
+        for base in bases:
+            annotations.update(base.__annotations__)
+            fields.extend(field.name for field in base._type_definition._fields)
+        for field in annotations:
+            if not field.startswith("_"):
+                annotations[field] = Optional[annotations[field]]
+        namespaces["__annotations__"] = annotations
+
+        klass = super().__new__(cls, name, bases, namespaces, **kwargs)
+        for field in fields:
+            setattr(klass, field, strawberry.UNSET)
```

Comment:
Probably this needs to first check if the klass already has defined a default value for the field